concordium-consensus/src/Concordium/External.hs

```diff
@@ -891,14 +891,17 @@ decodeBlockHash blockcstr = readMaybe <$> peekCString blockcstr
 decodeAccountAddress :: CString -> IO (Either String AccountAddress)
 decodeAccountAddress acctstr = addressFromBytes <$> BS.packCString acctstr
 
--- |Decode a null-terminated string as either an account address (base-58) or a
+-- |Decode a null-terminated string as either an account address (base-58), account index (Int) or a
 -- credential registration ID (base-16).
-decodeAccountAddressOrCredId :: CString -> IO (Maybe (Either CredentialRegistrationID AccountAddress))
+decodeAccountAddressOrCredId :: CString -> IO (Maybe AccountIdentifier)
 decodeAccountAddressOrCredId str = do
     bs <- BS.packCString str
     return $ case addressFromBytes bs of
-        Left _ -> Left <$> bsDeserializeBase16 bs
-        Right acc -> Just $ Right acc
+        Left _ -> Just $ 
+            case bsDeserializeBase16 bs of
+                Nothing -> AI $ read (BS.unpack bs)
```

Comment:
`read` is not a safe function, it can cause an exception. Use [readMaybe](https://hackage.haskell.org/package/base-4.16.1.0/docs/Text-Read.html#v:readMaybe) instead.

Reply:
:thumbsup: